### PR TITLE
Add mutual tls for Java based applications

### PR DESCRIPTION
**Pull request by faizanahmad055**
Signed-off-by: faizanahmad055 <faizan.ahmad55@outlook.com>

### CODE REVIEW

**Comment by stakater-user** [comment]
@faizanahmad055 Yikes! You better fix it before anyone else finds out! [Build](https://github.com/stakater-charts/application/commit/8bcc3cbcb70b4e2a99b721bf02b95e0457f252c6/checks) has Failed!

**Comment by stakater-user** [comment]
@faizanahmad055 Yikes! You better fix it before anyone else finds out! [Build](https://github.com/stakater-charts/application/commit/be4fd3fee26bd3e60f11918527b16e9f568538da/checks) has Failed!

File: application/templates/service.yaml [hunk 1]
```diff
@@ -1,4 +1,4 @@
-{{- if .Values.service }}
+{{- if or (.Values.service) (.Values.mutualTls.enabled) }}
```
Comment (hanzala1234) [inline application/templates/service.yaml:1]: why is that so?

File: application/templates/certificate.yaml [hunk 2]
```diff
@@ -18,12 +18,20 @@ metadata:
   name: {{ template "application.name" . }}-certificate
   namespace: {{ template "application.namespace" . }}
 spec:
+{{- if .Values.certificate.secretName }}
   secretName: {{ .Values.certificate.secretName }}
+{{- else }}
+  secretName: {{ template "application.name" . }}-client-tls
```
Comment (hanzala1234) [inline application/templates/certificate.yaml:24]: also I would suggest to take these in helpers.tpl. we just specify "certificate.secretName" here and conditions come from helpers.tpl. will be helpful to keep templates clean

File: application/templates/service.yaml (hunk 1 above)
Comment (hanzala1234) [reply application/templates/service.yaml:1]: we should not enable it when TLS is enabled. its like a requirement that in order to enable mTLS, service has to be enabled but this logic does not sound logical here

**Comment by rasheedamir** [comment]
closing as we are going to move this to umbrella chart and not in the leader chart

**Pull request closed without merging**